src/index.js

```diff
@@ -92,13 +92,20 @@ class CID {
     return this.buffer
   }
 
+  toV0String () {
+    if (this.version === 0) {
+      return mh.toB58String(this.multihash)
+    }
+
+    throw new Error(`Not supported on version: ${this.version}`)
+  }
+
   /* defaults to base58btc */
   toBaseEncodedString (base) {
     base = base || 'base58btc'
 
     switch (this.version) {
       case 0:
-        return mh.toB58String(this.multihash)
       case 1:
```

Comment:
Ah, good point, that was one of the things solved by by CIDV0 being always base58String, which is also defined by spec.

I believe the underlying problem, is that we want to be able to load CIDV0 with other types of encoding and what I'm reaching is that in order to avoid a lot of complexity added to other parts of the code, we should just force the user to have to update their CID to V1, if they want to expose another encoding, meaning that cid1.toBaseEncodedString('base16') would throw saying "must update to CID V1" and CID could easily have a method that does that (changing these methods to return new converted CIDS https://github.com/ipld/js-cid/blob/master/src/index.js#L87-L93).

Then, the other issue that needs to be solved is that this needs to be well communicated and the migration process should be as smooth as possible. Programmatically it is just a function call, but for all of those multihashes that are currently base58 encoded, I believe that having a tool at cid.io that converts them to a CIDv1 with the base encoding they want would be the best path.